scripts/irods/test/test_pam_password_authentication.py

```diff
@@ -603,67 +545,46 @@ def test_password_max_time_can_exceed_1209600__issue_3742_5096(self):
         original_max_time = self.admin.assert_icommand(
             ['iadmin', 'get_grid_configuration', self.configuration_namespace, max_time_option_name], 'STDOUT')[1].strip()
 
-        IrodsController().stop()
-
         auth_session_env_backup = copy.deepcopy(self.auth_session.environment_file_contents)
-        admin_session_env_backup = copy.deepcopy(self.admin.environment_file_contents)
         try:
-            service_account_environment_file_path = os.path.join(
-                os.path.expanduser('~'), '.irods', 'irods_environment.json')
-            with lib.file_backed_up(service_account_environment_file_path):
-                client_update = test_configurations.make_dict_for_ssl_client_environment(
-                    self.server_key_path, self.chain_pem_path, self.dhparams_pem_path)
-
-                lib.update_json_file_from_dict(service_account_environment_file_path, client_update)
+            client_update = {'irods_authentication_scheme': self.authentication_scheme}
+            self.auth_session.environment_file_contents.update(client_update)
 
-                self.admin.environment_file_contents.update(client_update)
+            # The test value is 2 hours more than the default in order to try a TTL value 1 greater and 1 less
+            # than the configured password_max_time while still remaining above 1209600 to show that there is
+            # nothing special about that value.
+            base_ttl_in_hours = 336 + 2
+            base_ttl_in_seconds = base_ttl_in_hours * 3600
 
-                client_update['irods_authentication_scheme'] = self.authentication_scheme
-                self.auth_session.environment_file_contents.update(client_update)
-
-                with temporary_core_file() as core:
-                    core.add_rule(test_configurations.get_pep_for_ssl(self.plugin_name))
-
-                    IrodsController().start(test_mode=True)
-
-                    # The test value is 2 hours more than the default in order to try a TTL value 1 greater and 1 less
-                    # than the configured password_max_time while still remaining above 1209600 to show that there is
-                    # nothing special about that value.
-                    base_ttl_in_hours = 336 + 2
-                    base_ttl_in_seconds = base_ttl_in_hours * 3600
-
-                    # Set password_max_time to the value for the test.
-                    option_value = str(base_ttl_in_seconds)
-                    self.admin.assert_icommand(
-                        ['iadmin', 'set_grid_configuration', self.configuration_namespace, max_time_option_name, option_value])
+            # Set password_max_time to the value for the test.
+            option_value = str(base_ttl_in_seconds)
+            self.admin.assert_icommand(
+                ['iadmin', 'set_grid_configuration', self.configuration_namespace, max_time_option_name, option_value])
```

Comment:
So these grid configuration options take affects as soon as they are changed, right?

Which is why the server doesn't need a config reload.

Reply:
Sort of. The configurations are set in the database immediately, and then are fetched from the database as needed. For this case, it's whenever a TTL value is passed when authenticating.

Reply:
We'll need to document that. The documentation effort is a push to be able to assert which configuration options are affected by SIGHUP and which are not.

Reply:
True. We should discuss that for sure. It's not yet clear which direction is better for admins/users.

Reply:
I think we can continue that conversation in https://github.com/irods/irods/issues/8348. If not, then we can open yet another issue to discuss the optimization piece.